src/main/java/com/adq/jenkins/xmljobtodsl/InitialArgumentsHandler.java

```diff
@@ -133,36 +140,56 @@ public static String getJobNameBasedOnPath(File file) {
 	}
 
 	public static String getJobPromotedBuildsXMLs(File file)
-		throws IOException {
+			throws IOException, SAXException, ParserConfigurationException {
 
 		IOUtils ioUtils = new IOUtils();
+		ArrayList<String> promotionNames = new ArrayList<>();
+
+		// parse xmlFile to get the names of promotion directories and add to list
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		DocumentBuilder db =  dbf.newDocumentBuilder();
+		Document doc = db.parse(file);
+
+		NodeList promotionList = doc.getDocumentElement().getElementsByTagName("activeProcessNames");
+
+		for (int i = 0; i < promotionList.getLength(); ++i) {
+			NodeList childList = promotionList.item(i).getChildNodes();
+			for (int j=0; j < childList.getLength(); j++) {
+				Node childNode = childList.item(j);
+				promotionNames.add(childNode.getTextContent());
+			}
+		}
 
 		File directoryPath = file.getAbsoluteFile().getParentFile();
 
 		String returnXML = "";
 
 		if(new File(directoryPath, "promotions").exists()) {
 			File promotionsPath = new File(directoryPath.getAbsolutePath() + "/promotions");
-			for (File promotionStepDir : promotionsPath.listFiles()) {
-				if (new File(promotionStepDir, "config.xml").exists()) {
-					String promotionConfigPath = promotionStepDir.getAbsolutePath() + "/config.xml";
-					// Get rid of the <?xml version='1.1' encoding='UTF-8'?> heading otherwise wont parse
-					String promotionStepXML = ioUtils.readFromFile(promotionConfigPath);
-					int endOfXMLHeadingIndex = promotionStepXML.indexOf("\n");
-
-					String versionHeadingRemovedXML = promotionStepXML.substring(endOfXMLHeadingIndex).trim();
-
-					// Insert name of job in between first line in promoted job and the rest of the file
-					int endOfNewSubstringXML = versionHeadingRemovedXML.indexOf("\n");
-					String firstHalfString = versionHeadingRemovedXML.substring(0, endOfNewSubstringXML).trim();
-					String secondHalfString = versionHeadingRemovedXML.substring(endOfNewSubstringXML).trim();
-					String buildStepName = String.format("<promotedBuildStepName>%s</promotedBuildStepName>\n", getJobNameBasedOnPath(new File(promotionConfigPath)));
-
-					returnXML += firstHalfString + buildStepName + secondHalfString;
+			File[] promotionPaths = promotionsPath.listFiles();
+			// for each promotion name, get the corresponding promotion path
+			for (Object promotionName : promotionNames) {
+				for (File promotionStepDir : promotionPaths) {
```

Comment:
Does Java let us do anything here to prevent the nested for loop? ie does promotionPaths contain promotionName or something like that

Also, why is the promotionName an object?